src/components/Navigation/NavItem/NavItem.tsx

```diff
@@ -9,18 +9,45 @@ export interface Props extends React.HTMLAttributes<HTMLDivElement> {
     pullRight?: boolean;
     /** add activeClassName to the rendered item. You should add this to react-routers NavLink */
     useActiveClass?: boolean;
+    /** React-Router version to determine how to handle useActiveClass */
+    routerVersion?: 5 | 6;
     /** Node to be rendered as a single navigation item. There should be only 1 child */
     children: React.ReactElement;
 }
 
+interface ChildProps extends Omit<Props, 'className' | 'children' | 'aria-current'> {
+    className?: string | (({ isActive }: { isActive: boolean }) => string);
+    activeClassName?: string;
+    'aria-current'?: string | boolean;
+    ref: React.ForwardedRef<HTMLElement>;
+}
+
 const { block, elem } = bem('NavItem', styles);
 
+/**
+ * NavItem supports React-Router's NavLink as a child. It makes sure we pass the correct class both
+ * in Active and Inactive states. Since the API for this changed between version 5 and 6, now you need to
+ * specify which router version you are using.
+ * For technical reasons only the usage of version 6 is demonstrated in storybook
+ */
 export const NavItem = React.forwardRef<HTMLElement, Props>((props, ref) => {
-    const { active, pullRight, useActiveClass, children, ...rest } = props;
+    const { active, pullRight, useActiveClass, routerVersion = 5, children, ...rest } = props;
     const ariaProp = active ? { 'aria-current': 'page' } : {};
-    const newProps = { ...ariaProp, ...rest, ...block(props), ref };
-    if (useActiveClass) {
+    const newProps: ChildProps = { ...ariaProp, ...rest, ref };
+    if (!useActiveClass) {
+        newProps.className = block(props).className;
+    } else if (routerVersion === 5) {
+        newProps.className = block(props).className;
         newProps.activeClassName = elem('active', props).className;
+    } else {
+        const activeClass = elem('active', props).className as string;
+        newProps.className = ({ isActive }) => {
+            const classArr: string[] = [block(props).className];
+            if (isActive) {
+                classArr.push(activeClass);
+            }
+            return classArr.join(' ');
+        };
```

Comment:
Now I'm thinking that probably was not a good idea to combine 2 versions in one component. There are a lot of `if` conditions and it's hard to review. But, let's keep it as it is.

Reply:
I hope we can remove support for v5 later, then it will look ok again